src/interfaces/parser.ts

```diff
@@ -112,6 +112,16 @@ export type FlagProps = {
   hidden?: boolean;
   required?: boolean;
   dependsOn?: string[];
+  relationships?: {
+    dependsOn?: {
+      type: 'all' | 'atLeastOne';
+      flags: string[];
```

Comment:
`flags: string[] | Array<{flag: string, value: any}>` would allow you to say things like:

flag A can't be used if flag B has value X (exclusive)
or
you can't use flag A unless flag B has value Y (dependsOn)

or even (allow or disallow multiple values from a flag)
```
[{ flag: 'a', value: 'x'}, {flag: 'a', value: 'y'}]
```